src/docc/plugins/python/cst.py

```diff
@@ -560,8 +560,15 @@ def enter_class_def(self, node: CstNode, cst_node: cst.ClassDef) -> Visit:
         self.old_stack.append(class_context)
         self.old_stack.append(body_context)
 
-        for cst_statement in cst_node.body.body:
-            self.old_stack[-1].child_offset += 1
+        for idx, cst_statement in enumerate(cst_node.body.body):
+            self.old_stack[-1].child_offset = idx
```

Comment:
Is this a stylistic change, or is this important for fixing the issue? If it's the latter, can you briefly explain what went wrong here? When would `+= 1` be different from `idx`?

Even if it's just stylistic, it's definitely an improvement.

Reply:
I noticed that the `child_offset` was being incremented also at `exit()` L843, which resulted in statements being assigned offsets in multiples of 2. so used `idx` to assign it to its true position